app/views/layouts/application.html.haml

```diff
@@ -29,11 +29,11 @@
     %meta{ content: asset_pack_path('media/images/govuk-opengraph-image.png'), property: "og:image" }/
     = csrf_meta_tags
   %body{ class: body_class, data: { vacancy_state: @vacancy.present? ? @vacancy.state : 'create' } }
-    = render partial: 'shared/google_tag_manager_body'
+    = render partial: 'shared/google_tag_manager_body' if !CookiesBannerFeature.enabled? || cookies['consented-to-cookies'] == 'yes'
     :javascript
       document.body.className = ((document.body.className) ? document.body.className + ' js-enabled' : 'js-enabled');
     = render 'shared/skip_links'
-    = render 'layouts/cookies_banner' if CookiesBannerFeature.enabled? && !current_page?(page_path('cookies'))
+    = render 'layouts/cookies_banner' unless !CookiesBannerFeature.enabled? || controller_name == 'cookies_preferences' || cookies_preference_set?
```

Comment:
I don't know how much time it takes for an app to check a little variable for true/false (probably almost none), but we might shave off a millisecond or two by putting the most-frequently-true thing to evaluate first (`cookies_preference_set?`).
🤷

Reply:
The order is important. 

If the feature is disabled, **don't render**
If the feature is enabled, but we're at the cookies_preferences create/new action, **don't render**
If the feature is enabled, and we're not at the cookies_preferences create/new action, but cookies preferences are set, **don't render**
If the feature is enabled, and we're not at the cookies_preferences create/new action, and cookies preferences are not set, **render**